src/main/java/org/fhirvp/usecase/impl/builder/FHIRResourceBuilderUseCaseImpl.java

```diff
@@ -0,0 +1,178 @@
+                                                .value("http://fhir.de/CodeSystem/ifa/pzn")
+                                                .build())
+                                        .code(Code.builder()
+                                                .value(vaccineCode)
+                                                .build())
+                                        .build())
+                                .build())
+                        .targetDisease(CodeableConcept.builder()
+                                .coding(Coding.builder()
+                                        .system(Uri.builder()
+                                                .value("http://hl7.org/fhir/sid/icd-10")
+                                                .build())
+                                        .code(Code.builder()
+                                                .value(targetDiseaseCode)
+                                                .build())
+                                        .build())
+                                .build())
+                        .forecastStatus(CodeableConcept.builder()
+                                .coding(Coding.builder()
+                                        .system(Uri.builder()
+                                                .value("http://terminology.hl7.org/CodeSystem/immunization-recommendation-status")
+                                                .build())
+                                        .code(Code.builder()
+                                                .value(forecastStatus.name().toLowerCase())
+                                                .build())
+                                        .build())
+                                .build())
+                        .dateCriterion(dateCriteria)
+                        .supportingImmunization(supportingImmunizations)
+                        .build())
+                .extension(extensions)
+                .build();
+    }
+
+    private ImmunizationRecommendation.Recommendation.DateCriterion buildImmunizationRecommendationDateCriterion(String loincCode, Calendar date) {
+        return ImmunizationRecommendation.Recommendation.DateCriterion.builder()
+                .code(CodeableConcept.builder()
+                        .coding(Coding.builder()
+                                .system(Uri.builder()
+                                        .value("http://loinc.org")
+                                        .build())
+                                .code(Code.builder()
+                                        .value(loincCode)
+                                        .build())
+                                .build())
+                        .build())
+                .value(DateTime.of(LocalDate.of(date.get(Calendar.YEAR), date.get(Calendar.MONTH), date.get(Calendar.DAY_OF_MONTH))))
+                .build();
+    }
+
+    private Reference buildSupportingImmunization(String id) {
```

Comment:
You are not using the `id` parameter in the method body.

Reply:
Addressed at a9b6f4d